### DISCUSSION TITLE
X-axis missing first label (and sometimes last label!)

### DISCUSSION THREAD

**Discussion by cokecan72** [RomRider/apexcharts-card#606] [Q&A]
Hi Guys

Have been seeing a weird issue where some of the charts I'm working on are missing the first label on the X-Axis. I've attached an image showing the issue. I know this chart has been customised a fair bit to display this way, but even as a super basic version, the first label is still missing.

The chart in question is a column type chart that shows 7x days worth of total electricity usage (current usage for today and 6x previous days). 

I'm using this to group the data from a sensor that just shows daily usage:

` group_by:
              func: last
              duration: 23h59m59s`

Interestingly, if I use 1d as the duration, the last label is often missing. I'm guessing this is something to do with when the last data was received on the last day? Element inspector suggests the label just isn't there on the webpage.

Does anyone have any idea how to get the first label showing?

<img width="507" alt="Screenshot 2023-09-19 at 2 18 47 pm" src="https://github.com/RomRider/apexcharts-card/assets/109978326/cc4c4643-8604-4f03-8cb3-22363dd5a708">

**Comment by cokecan72** [comment]
For the remaining part of yesterday since my initial post, I did get that first X-axis label displaying correctly! I did this by adding an offset of "-1m" to the span of the graph. Today however, we are back to the first label missing.

I've double checked the data being reported by the sensor I'm using and it's reporting "live" data right up to 23:59:59, but given what I'm seeing with the Duration and Offset settings it's making me wonder if this is something weird about how Apexcharts is reading and grouping the data? The fact that yesterday afternoon at least I've had to set the duration to 23h59m59s instead of just 1d and that I had to add a -1m offset to the span?!

**Comment by evlo** [comment]
did you ever figure out workaround?

**Reply by cokecan72** [reply]
Hey, yes I did but it's a bit messy. I raised this as a bug which was closed without resolution except my work around here: https://github.com/RomRider/apexcharts-card/issues/639

I think whats happening is that when the Y-axis is hidden, it breaks something when the X-axis label is set to ddd. My workaround involves setting it to display the Y-axis in the yaxis options and then hiding it after using CSS with the Lovelace Card-Mod. From my testing, you can hide the Y-axis in the yaxis options providing you set the X-axis format to dd or something else and this will display fine. If you want the X-axis format as ddd like I do then you got to show the Y-axis in yaxis options and then hide it using CSS
